traits/has_traits.py

```diff
@@ -416,14 +416,24 @@ def _property_method ( class_dict, name ):
 _HasTraits = None
 
 class MetaHasTraits ( type ):
-    ### JMS: Need a docstring here.
+    """ Controls the creation of HasTraits classes.
+
+    The heavy work is done by the `create_traits_meta_dict` function, which
+    takes the ``class_dict`` dictionary of class members and extracts and
+    processes the trait declarations in it. The trait declarations are then
+    added back to the class dictionary and passed off to the __new__ method
+    of the type superclass, to be added to the class.
+
+    """
     # All registered class creation listeners.
     #
     # { Str class_name : Callable listener }
     _listeners = {}
 
     def __new__ ( cls, class_name, bases, class_dict ):
-        mhto = MetaHasTraitsObject( cls, class_name, bases, class_dict, False )
+        traits_meta_dict = create_traits_meta_dict(
+            class_name, bases, class_dict, False )
+        class_dict.update(traits_meta_dict)
```

Comment:
> this function is being used by the category machinery

Yes, that's exactly why I took that tortuous route. I agree with having a clean function for `MetaHasTraits`, especially since the `Category` code seems on its way out.